target_chains/solana/programs/pyth-solana-receiver/tests/common/mod.rs

```diff
@@ -0,0 +1,227 @@
+            ema_price:         14,
+            ema_conf:          15,
+        },
+    ]
+}
+
+pub fn dummy_price_updates() -> (MerkleTree<Keccak160>, Vec<MerklePriceUpdate>) {
+    let price_feed_messages: Vec<Message> = dummy_price_messages()
+        .iter()
+        .map(|x| Message::PriceFeedMessage(*x))
+        .collect::<Vec<Message>>();
+    let price_feed_message_as_vec: Vec<Vec<u8>> = price_feed_messages
+        .iter()
+        .map(|x| pythnet_sdk::wire::to_vec::<_, byteorder::BigEndian>(&x).unwrap())
+        .collect();
+
+    let merkle_tree_accumulator =
+        MerkleTree::<Keccak160>::from_set(price_feed_message_as_vec.iter().map(|x| x.as_ref()))
+            .unwrap();
+    let merkle_price_updates: Vec<MerklePriceUpdate> = price_feed_message_as_vec
+        .iter()
+        .map(|x| MerklePriceUpdate {
+            message: PrefixedVec::<u16, u8>::from(x.clone()),
+            proof:   merkle_tree_accumulator.prove(x.as_ref()).unwrap(),
+        })
+        .collect();
+
+    (merkle_tree_accumulator, merkle_price_updates)
+}
+
+
+pub async fn build_merkle_root_encoded_vaa(
+    merkle_tree_accumulator: MerkleTree<Keccak160>,
+    data_source: &DataSource,
+) -> Account {
+    let merkle_tree_payload: Vec<u8> = merkle_tree_accumulator.serialize(1, 1);
+
+    let vaa_header: Vaa<Box<RawMessage>> = Vaa {
+        version:            1,
+        guardian_set_index: 0,
+        signatures:         vec![],
+        timestamp:          0,
+        nonce:              0,
+        emitter_chain:      Chain::from(data_source.chain),
+        emitter_address:    wormhole_sdk::Address(data_source.emitter.to_bytes()),
+        sequence:           0,
+        consistency_level:  0,
+        payload:            <Box<RawMessage>>::from(merkle_tree_payload),
+    };
+
+    let encoded_vaa_data = (
+        <EncodedVaa as anchor_lang::Discriminator>::DISCRIMINATOR,
+        Header {
+            status:          ProcessingStatus::Verified,
+            write_authority: Pubkey::new_unique(),
+            version:         1,
+        },
+        serde_wormhole::to_vec(&vaa_header).unwrap(),
+    )
+        .try_to_vec()
+        .unwrap();
+
+
+    Account {
+        lamports:   Rent::default().minimum_balance(encoded_vaa_data.len()),
+        data:       encoded_vaa_data,
+        owner:      BRIDGE_ID,
+        executable: false,
+        rent_epoch: 0,
+    }
+}
+
+
+pub struct ProgramTestFixtures {
+    pub program_simulator:    ProgramSimulator,
+    pub encoded_vaa_address:  Pubkey,
+    pub merkle_price_updates: Vec<MerklePriceUpdate>,
+}
+/**
+ * Setup to test the Pyth Receiver. The return values are a tuple composed of :
+ * - The program simulator, which is used to send transactions
+ * - The pubkey of an encoded VAA account, which is pre-populated and can be used to test post_updates
+ * - A vector of MerklePriceUpdate, corresponding to that VAA
+ */
+pub async fn setup_pyth_receiver() -> ProgramTestFixtures {
```

Comment:
this seems like an odd combination of functionality. It seems like most tests will want to send their own set of price updates (?) Will this ever be reused?

Reply:
(it's a little annoying that you need to set up the encoded vaa accounts before the simulator gets created. Tests probably want the ability to modify the set of encoded VAA accounts themselves. Not sure if there's a good way to facilitate that though.

Even if you can't address that, passing in the set of accounts/merkle trees here would make this code more flexible)

Reply:
I'll think about this in future development